src/services/vision/client.ts

```diff
@@ -0,0 +1,149 @@
+
+/**
+ * A gRPC-web client for a Vision service.
+ *
+ * @group Clients
+ */
+export class VisionClient implements Vision {
+  private client: VisionServiceClient;
+  private readonly name: string;
+  private readonly options: Options;
+
+  constructor(client: RobotClient, name: string, options: Options = {}) {
+    this.client = client.createServiceClient(VisionServiceClient);
+    this.name = name;
+    this.options = options;
+  }
+
+  private get service() {
+    return this.client;
+  }
+
+  async getDetectionsFromCamera(cameraName: string, extra: StructType = {}) {
+    const { service } = this;
+
+    const request = new pb.GetDetectionsFromCameraRequest();
+    request.setName(this.name);
+    request.setCameraName(cameraName);
+    request.setExtra(Struct.fromJavaScript(extra));
+
+    this.options.requestLogger?.(request);
+
+    const response = await promisify<
+      pb.GetDetectionsFromCameraRequest,
+      pb.GetDetectionsFromCameraResponse
+    >(service.getDetectionsFromCamera.bind(service), request);
+
+    return response.getDetectionsList().map((x) => x.toObject());
+  }
+
+  async getDetections(
+    image: Uint8Array,
+    width: number,
+    height: number,
+    mimeType: MimeType,
+    extra: StructType = {}
+  ) {
+    const { service } = this;
+
+    const request = new pb.GetDetectionsRequest();
+    request.setName(this.name);
+    request.setImage(image);
+    request.setWidth(width);
+    request.setHeight(height);
+    request.setMimeType(mimeType);
+    request.setExtra(Struct.fromJavaScript(extra));
+
+    this.options.requestLogger?.(request);
+
+    const response = await promisify<
+      pb.GetDetectionsRequest,
+      pb.GetDetectionsResponse
+    >(service.getDetections.bind(service), request);
+
+    return response.getDetectionsList().map((x) => x.toObject());
+  }
+
+  async getClassificationsFromCamera(
+    cameraName: string,
+    count: number,
+    extra: StructType = {}
+  ) {
+    const { service } = this;
+
+    const request = new pb.GetClassificationsFromCameraRequest();
+    request.setName(this.name);
+    request.setCameraName(cameraName);
+    request.setN(count);
+    request.setExtra(Struct.fromJavaScript(extra));
+
+    this.options.requestLogger?.(request);
+
+    const response = await promisify<
+      pb.GetClassificationsFromCameraRequest,
+      pb.GetClassificationsFromCameraResponse
+    >(service.getClassificationsFromCamera.bind(service), request);
+
+    return response.getClassificationsList().map((x) => x.toObject());
+  }
+
+  async getClassifications(
+    image: Uint8Array,
+    width: number,
+    height: number,
+    mimeType: MimeType,
+    count: number,
+    extra: StructType = {}
+  ) {
+    const { service } = this;
+
+    const request = new pb.GetClassificationsRequest();
+    request.setName(this.name);
+    request.setImage(image);
+    request.setWidth(width);
+    request.setHeight(height);
+    request.setMimeType(mimeType);
```

Comment:
I don't think these are necessary - they should be able to be read from the headers as the images, as long as the images are of a known type. It would be better to just have a helper function that could be used to read image headers. This is a common task and there are probably functions already out there